cognite/client/data_classes/workflows.py

```diff
@@ -317,6 +349,7 @@ def __init__(
         self,
         external_id: str,
         parameters: WorkflowTaskParameters,
+        type: Literal["function", "transformation", "cdf", "dynamic"] | None = None,
```

Comment:
This is double information that does not have to be exposed to the user. The type is kept as the class type

Reply:
Can't you just use a property then? 
```python
@property
def type(self) -> str:
    self.paramterers.task_type
```

The way the class is set up now you can create a `WorkflowTask` which has `parameters` of type `FunctionTaskParamters` while you set the type to `transformation`. Having an option to create an inconsistent instance is a code smell to me, as it means you have to ensure you input the same information twice.